lib/HLSL/DxilContainerReflection.cpp

```diff
@@ -272,9 +272,11 @@ namespace hlsl {
 HRESULT CreateDxilShaderReflection(const DxilProgramHeader *pProgramHeader, const DxilPartHeader *pRDATPart, REFIID iid, void **ppvObject) {
   if (!ppvObject)
     return E_INVALIDARG;
+  PublicAPI api = DxilShaderReflection::IIDToAPI(iid);
```

Comment:
Nit, shouldn't this be moved down lower so that api is closer to where it's used, sort of like how the code was originally?

Reply:
We want to recognize an invalid IID early and `return E_INVALIDARG` before creating the object.  Otherwise, it will happily accept any IID and treat unrecognized ones as if they match `ID3D12ShaderReflection` (as it did before).